### PR TITLE
Fix legend edgecolor face

### LINKED ISSUE #20258: Using edgecolors='face' with stackplot causes value error when using plt.legend()
### Bug report

I was making a stack plot and wanted the edge colors of the plotted lines to be the same as the fill color instead of the default white, so I set edgecolor='face'. This produces a value error when using plt.legend(). Without the legend function, there are no errors and it plots as expected. I tried changing different parameters of the legend() function, but didn't have any success.

**Code for reproduction**

```python
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
# Create a dataset
my_count = [
    "France", "Australia", "Japan", "USA", "Germany",
    "Congo", "China", "England", "Spain", "Greece",
    "Marocco", "South Africa", "Indonesia", "Peru", "Chili", "Brazil"
]
df = pd.DataFrame(
    {
        "country": np.repeat(my_count, 10),
        "years": list(range(2000, 2010)) * 16,
        "value": np.random.rand(160)
    }
)
pivot = df.groupby(["country", "years"])["value"].sum().unstack(0).cumsum()
# Plot dataset
plt.stackplot(
    pivot.index.values,
    pivot.T.values,
    labels=pivot.columns,
    alpha=0.2,
    edgecolors="face"
)
plt.legend()
plt.show()
```

**Actual outcome**

image
with the following value error:

```
ValueError: Using a string of single character colors as a color sequence is not supported. The colors can be passed as an explicit list instead.
```

**Expected outcome**

The same graph as above, but with a legend.

**Matplotlib version**
  * Operating system: Microsoft Windows 10
  * Matplotlib version: 3.4.2 
  * Matplotlib backend: module://ipykernel.pylab.backend_inline 
  * Python version: 3.8.5
  * Other libraries: numpy, pandas (to create dataset)

Installed matplotlib using 'pip install matplotlib'.

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Closes #20258

Passing `'face'` to a polygon that would show up in the legend would not be recognized.  Now it is...

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
ax.fill_between([0, 1, 2], [1, 2, 3], [2, 3, 4], 
                facecolor='r', edgecolor='face', label='Fill')
ax.legend()
plt.show()
```

#### before:

```
ValueError: Using a string of single character colors as a color sequence is not supported. The colors can be passed as an explicit list instead.
```

### Now:

face

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/legend_handler.py [hunk 1]
```diff
@@ -749,6 +749,9 @@ def get_first(prop_array):
                 return None
         edgecolor = getattr(orig_handle, '_original_edgecolor',
                             orig_handle.get_edgecolor())
+        if edgecolor == 'face':
```
Comment (jklymak) [inline lib/matplotlib/legend_handler.py:752]: Note we can't just get rid of the line before as it causes some hatching tests to fail for some reason...

**Comment by efiring** [comment]
#8575 put in the mysterious fix for the hatching. Strangely, although `edgecolor="face"` was failing, `edgecolor="none"` was not.

**Comment by jklymak** [comment]
Thanks - we could also do 
```
edgecolor = orig_handle.get_edgecolor()
if isempty(edgecolor):
   edgecolor = None;
```
Or fix it in `collections.py:get_edgecolor`

**Comment by jklymak** [comment]
Second commit checks for no colors...

**Comment by jklymak** [comment]
Close in favour of #20265

**Pull request closed without merging**